Library/Homebrew/os/mac/sdk.rb

```diff
@@ -55,5 +52,43 @@ def sdk_paths
         end
       end
     end
+
+    class XcodeSDKLocator < BaseSDKLocator
+      private
+
+      def sdk_prefix
+        @sdk_prefix ||= begin
+          # Xcode.prefix is pretty smart, so let's look inside to find the sdk
+          sdk_prefix = "#{Xcode.prefix}/Platforms/MacOSX.platform/Developer/SDKs"
+          # Xcode < 4.3 style
+          sdk_prefix = "/Developer/SDKs" unless File.directory? sdk_prefix
+          # Finally query Xcode itself (this is slow, so check it last)
+          sdk_prefix = File.join(Utils.popen_read(DevelopmentTools.locate("xcrun"), "--show-sdk-platform-path").chomp, "Developer", "SDKs") unless File.directory? sdk_prefix
+
+          sdk_prefix
+        end
+      end
+    end
+
+    class CLTSDKLocator < BaseSDKLocator
+      private
+
+      # While CLT SDKs existed prior to Xcode 10, those packages also
+      # installed a traditional Unix-style header layout and we prefer
+      # using that
+      # As of Xcode 10, the Unix-style headers are installed via a
+      # separate package, so we can't rely on their being present.
+      # This will only look up SDKs on Xcode 10 or newer, and still
+      # return nil SDKs for Xcode 9 and older.
+      def sdk_prefix
+        @sdk_prefix ||= begin
+          if !CLT.separate_header_package?
```

Comment:
I don't think we should favor the header package even when installed because it's missing stuff.

Reply:
Sorry if that wasn't clear. Since the "header package" is only a thing for Xcode 10, I was only talking about Xcode 10.

Reply:
> I don't think we should favor the header package even when installed because it's missing stuff.

👍 

> Pre-Xcode 10, I think we should still use the headers in /usr. That's how we've done it before, and we might as well keep consistency there.

👍 

> so I think this ends up favoring /usr/include on 10.13 with Xcode 10/CLT 10. Is that intentional?

If it is: I don't think it should be. I think that should also check the Xcode version >= 10.

Reply:
> I think that should also check the Xcode version >= 10.

Fixed. This now favours the SDK with Xcode 10 on 10.13.

> I don't think we should favor the header package even when installed because it's missing stuff.

With the log update for `CLT.separate_header_package?` this will now only use the CLT SDK for Xcode 10 onwards, and it will always use it on Xcode 10.